src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -360,52 +360,46 @@ private static void AddProjectFileDependenciesForPackageReference(PackageSpec pr
+                    .Where(graphItem => graphItem.IsCentralTransitive && centralPackageVersionsForFramework?.ContainsKey(graphItem.Key.Name) == true)
+                    .Select((graphItem) =>
+                    {
+                        CentralPackageVersion matchingCentralVersion = centralPackageVersionsForFramework[graphItem.Key.Name];
+                        Dictionary<string, LibraryIncludeFlags> dependenciesIncludeFlags = _includeFlagGraphs[targetGraph];
 
-                    // The transitive dependencies enforced by the central package version management file are written to the assets to be used by the pack task.
-                    IEnumerable<LibraryDependency> centralEnforcedTransitiveDependencies = targetGraph
-                        .Flattened
-                        .Where(graphItem => graphItem.IsCentralTransitive && centralPackageVersionsForFramework?.ContainsKey(graphItem.Key.Name) == true)
-                        .Select((graphItem) =>
+                        var libraryDependency = new LibraryDependency()
                         {
-                            CentralPackageVersion matchingCentralVersion = centralPackageVersionsForFramework[graphItem.Key.Name];
+                            LibraryRange = new LibraryRange(matchingCentralVersion.Name, matchingCentralVersion.VersionRange, LibraryDependencyTarget.Package),
+                            ReferenceType = LibraryDependencyReferenceType.Transitive,
+                            VersionCentrallyManaged = true,
+                            IncludeType = dependenciesIncludeFlags[matchingCentralVersion.Name]
+                        };
 
-                            Debug.Assert(dependenciesIncludeFlags.ContainsKey(matchingCentralVersion.Name));
+                        return libraryDependency;
+                    });
 
-                            var libraryDependency = new LibraryDependency()
-                            {
-                                LibraryRange = new LibraryRange(matchingCentralVersion.Name, matchingCentralVersion.VersionRange, LibraryDependencyTarget.Package),
-                                ReferenceType = LibraryDependencyReferenceType.Transitive,
-                                VersionCentrallyManaged = true,
-                                IncludeType = dependenciesIncludeFlags[matchingCentralVersion.Name]
-                            };
-                            
-                            return libraryDependency;
-                        });
-
-                    if (centralEnforcedTransitiveDependencies.Any())
-                    {
-                        var centralEnforcedTransitiveDependencyGroup = new CentralTransitiveDependencyGroup
-                                (
-                                    targetGraph.Framework,
-                                    centralEnforcedTransitiveDependencies
-                                );
+                if (centralEnforcedTransitiveDependencies.Any())
```

Comment:
Should we add a transitive dependency group for every CPVM enabled project?

Allows us to differentiate between `CPVM is ignoring things` and `CPVM did not find anything`. 

Don't consider this a blocker though.

Reply:
Added issue https://github.com/NuGet/Home/issues/9409 to track.